src/js/components/search/filters/location/LocationSearch.jsx

```diff
@@ -29,15 +100,21 @@ export default class LocationSearch extends React.Component {
             <div className="location-filter search-filter">
                 <CountryType
                     {...this.props}
-                    toggleCountry={this.props.toggleCountry}
-                    locationOption={this.props.locationOption} />
+                    toggleCountry={this.toggleCountry}
+                    locationOption={this.state.locationOption} />
                 <LocationListContainer
-                    selectedLocations={this.props.selectedLocations}
+                    {...this.props}
                     selectLocation={this.props.selectLocation}
-                    locationOption={this.props.locationOption} />
+                    queryAutocompleteLocations={this.queryAutocompleteLocations} />
                 {selectedLocations}
             </div>
         );
     }
 }
+
+export default connect(
```

Comment:
This is now a container, not a component. The container-related code (anything that has to do with API calls or manipulating data that's not directly display-related) needs to be moved to its parent container (`LocationSearchContainer`). Components are not allowed to be connected to Redux directly.

Please refactor.